### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by zeke**
This PR moves the majority of the code into a standalone module, and adds some tests for it using jest.

I know that the ideal solution would be to use mock data and not hit the network, but at least this is a start. `GH_TOKEN` should be present on Travis already since we've set up semantic-release here.

@gr2m let me know what you think.

### CODE REVIEW

**Comment by zeke** [comment]
Just realized one disadvantage of this approach is that PRs from outside contributors will fail on CI, because they won't have `GH_TOKEN` in the environment. 🤔

File: test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  })
+
+  xtest('creates a nice title for named releases', async () => {
+    const profile = await profileRelease({
+      owner: 'probot',
+      repo: 'probot',
+      tagName: 'v6.0.0'
+    })
+    expect(profile.currentRelease.title).toEqual('v6.0.0 – 2018-02-28')
+  })
```
Comment (gr2m) [inline test.js]: did you make this `xtest` on purpose?

**Review by gr2m — APPROVED** [review]
🚢

File: test.js (hunk 1 above)
Comment (zeke) [reply test.js]: Yeah. Was timing out and didn't want to bother with it!

**Pull request merged**